C#: Exported games fail to load hostfxr

### Godot version

1f9b992d85fb974c7aa1fadaf2d1b4ff59850137

### System information

Windows 10

### Issue description

Hey 😄 

With the mono module, exported games crash as soon as they start. This happens with templates compiled in `release` and in `release_debug`. My editor itself is built in `debug`.  
Running them through the console script yields the following errors.
```
ERROR: Can't open dynamic library: C:/Users/paulloz/Desktop/Builds/Sandbox/data_Sandbox-G4NET6/hostfxr.dll, error: Error 193: %1 is not a valid Win32 application.
   at: (platform\windows\os_windows.cpp:257)
ERROR: Can't open dynamic library: C:/Users/paulloz/Desktop/Builds/Sandbox/data_Sandbox-G4NET6/Sandbox-G4NET6.dll, error: Error 193: %1 is not a valid Win32 application.
   at: (platform\windows\os_windows.cpp:257)
ERROR: .NET: Failed to load hostfxr
   at: (modules\mono\mono_gd\gd_mono.cpp:435)
```

I believe @raulsntos was saying the same kind of thing occurs on Linux regarding hostfxr. However, the two first errors here may not have their counterpart on other platforms.  

I must note that specifically on Windows 10 (x86_64), exports templates weren't compiling at first. I had to apply the following patch, that I'm not confident is valid, especially the second part.

```diff
diff --git a/modules/mono/mono_gd/gd_mono.cpp b/modules/mono/mono_gd/gd_mono.cpp
index 0532cc915b..16bbc6a672 100644
--- a/modules/mono/mono_gd/gd_mono.cpp
+++ b/modules/mono/mono_gd/gd_mono.cpp
@@ -270,7 +270,7 @@ load_assembly_and_get_function_pointer_fn initialize_hostfxr_self_contained(
        int i = 1;
        for (const String &E : cmdline_args) {
                HostFxrCharString &stored = argv_store.push_back(str_to_hostfxr(E))->get();
-               argv.write[i] = stored.ptr();
+               argv.write[i] = (const char_t *)stored.ptr();
                i++;
        }

@@ -356,7 +356,7 @@ godot_plugins_initialize_fn initialize_hostfxr_and_godot_plugins(bool &r_runtime
        print_verbose(".NET: hostfxr initialized");

        int rc = load_assembly_and_get_function_pointer(get_data(assembly_path),
-                       str_to_hostfxr("GodotPlugins.Game.Main, " + assembly_name),
+                       get_data(str_to_hostfxr("GodotPlugins.Game.Main, " + assembly_name)),
                        HOSTFXR_STR("InitializeFromGameProject"),
                        UNMANAGEDCALLERSONLY_METHOD,
                        nullptr,
```
This is an attempt to fix the following compilation errors.
```
modules\mono\mono_gd\gd_mono.cpp(273): error C2440: '=': cannot convert from 'const char16_t *' to 'T'
        with
        [
            T=const char_t *
        ]
modules\mono\mono_gd\gd_mono.cpp(273): note: Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast)
modules\mono\mono_gd\gd_mono.cpp(359): error C2664: 'int (const char_t *,const char_t *,const char_t *,const char_t *,void *,void **)': cannot convert argument n from '`anonymous-namespace'::HostFxrCharString' to 'const char_t *'
modules\mono\mono_gd\gd_mono.cpp(359): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
```

### Steps to reproduce

Should be really easy to reproduce:
* Create a project
* Add a C# script
* Export
* Run the exported game

### Minimal reproduction project

_No response_

--- a/modules/mono/editor/GodotTools/GodotTools/Build/BuildSystem.cs
+++ b/modules/mono/editor/GodotTools/GodotTools/Build/BuildSystem.cs
@@ -176,8 +176,9 @@ namespace GodotTools.Build
                 arguments.Add("--no-restore");
 
             // Incremental or rebuild
-            if (buildInfo.Rebuild)
-                arguments.Add("--no-incremental");
+            // TODO: Not supported in `dotnet publish` (https://github.com/dotnet/sdk/issues/11099)
+            // if (buildInfo.Rebuild)
+            //     arguments.Add("--no-incremental");
 
             // Configuration
             arguments.Add("-c");

--- a/modules/mono/editor/GodotTools/GodotTools/Export/ExportPlugin.cs
+++ b/modules/mono/editor/GodotTools/GodotTools/Export/ExportPlugin.cs
@@ -112,7 +112,7 @@ namespace GodotTools.Export
             string buildConfig = isDebug ? "ExportDebug" : "ExportRelease";
 
             // TODO: This works for now, as we only implemented support for x86 family desktop so far, but it needs to be fixed
-            string arch = features.Contains("64") ? "x86_64" : "x86";
+            string arch = features.Contains("x86_64") ? "x86_64" : "x86";
 
             string ridOS = DetermineRuntimeIdentifierOS(platform);
             string ridArch = DetermineRuntimeIdentifierArch(arch);

--- a/modules/mono/editor/GodotTools/GodotTools/Utils/OS.cs
+++ b/modules/mono/editor/GodotTools/GodotTools/Utils/OS.cs
@@ -70,9 +70,7 @@ namespace GodotTools.Utils
         {
             ["Windows"] = Platforms.Windows,
             ["macOS"] = Platforms.MacOS,
-            ["LinuxBSD"] = Platforms.LinuxBSD,
-            // "X11" for compatibility, temporarily, while we are on an outdated branch
-            ["X11"] = Platforms.LinuxBSD,
+            ["Linux"] = Platforms.LinuxBSD,
             ["UWP"] = Platforms.UWP,
             ["Haiku"] = Platforms.Haiku,
             ["Android"] = Platforms.Android,

--- a/modules/mono/mono_gd/gd_mono.cpp
+++ b/modules/mono/mono_gd/gd_mono.cpp
@@ -103,12 +103,12 @@ const char_t *get_data(const HostFxrCharString &p_char_str) {
 }
 
 #ifdef TOOLS_ENABLED
-String find_hostfxr(size_t p_known_buffet_size, get_hostfxr_parameters *p_get_hostfxr_params) {
+String find_hostfxr(size_t p_known_buffer_size, get_hostfxr_parameters *p_get_hostfxr_params) {
 	// Pre-allocate a large buffer for the path to hostfxr
 	Vector<char_t> buffer;
-	buffer.resize(p_known_buffet_size);
+	buffer.resize(p_known_buffer_size);
 
-	int rc = get_hostfxr_path(buffer.ptrw(), &p_known_buffet_size, p_get_hostfxr_params);
+	int rc = get_hostfxr_path(buffer.ptrw(), &p_known_buffer_size, p_get_hostfxr_params);
 
 	ERR_FAIL_COND_V_MSG(rc != 0, String(), "get_hostfxr_path failed with code: " + itos(rc));
 

